### PR TITLE
Quality of life improvements for formula handling around BigDecimal columns.

### PR DESCRIPTION
**Pull request by jcferretti**
Deephaven - Google Chrome_143

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java [hunk 1]
```diff
@@ -19428,4 +19430,436 @@ public static int negate(short a) {
...
+
+    public static BigDecimal multiply(BigDecimal a, long b) {
+        if (a == null || b == QueryConstants.NULL_LONG) {
+            return null;
+        }
+        return a.multiply(BigDecimal.valueOf(b));
+    }
+
+    public static BigDecimal multiply(long a, BigDecimal b) {
+        if (a == QueryConstants.NULL_LONG || b == null) {
```
Comment (cpwright) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java:19498]: should you really repeat checks if you are just going to use the commutative operator?

File: engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java [hunk 2]
```diff
@@ -19428,4 +19430,436 @@ public static int negate(short a) {
...
+
+        if (b == QueryConstants.NULL_DOUBLE) {
+            return 1;
+        }
+
+        return a.compareTo(BigDecimal.valueOf(b));
+    }
+
+    public static int compareTo(double a, BigDecimal b) {
+        if (a == QueryConstants.NULL_DOUBLE) {
```
Comment (cpwright) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java:19703]: Deephaven normally has special comparisons for Double.NaN.

**Review by cpwright — CHANGES_REQUESTED** [review]
This all needs unit test coverage; we can't just merge it unless we are sure that we cover all these cases from an update or where statement.

**Review by cpwright — COMMENTED** [review]
So I think you did a good job covering the functions.  There is a question as to whether it makes sense to verify a few uses in a formula.  I know you have it in the demo, but a handful of unit tests (less exhaustive) is still probably worth it to ensure we don't break e.g. the parser's handling of these things.

**Review by cpwright — DISMISSED** [review]

**Review by cpwright — APPROVED** [review]

**Comment by jcferretti** [comment]
Part of #1753

**Pull request merged**